frontend/apps/cloudstorage/dropbox.lua

```diff
@@ -10,9 +10,280 @@ local util = require("util")
 local Screen = require("device").screen
 local T = require("ffi/util").template
 local _ = require("gettext")
+local lfs = require("libs/libkoreader-lfs")
 
 local DropBox = {}
 
+function DropBox:showFilesAndFolders(url, password)
+    return DropBoxApi:showFilesAndFoldersV2(url, password)
+end
+
+function DropBox:getRemoteFilesMap(url,password,treePath)
+    local tree = {}
+    local UI = require("ui/trapper")
+    local files = DropBox:showFilesAndFolders(url,password)
+    for i,file in ipairs(files) do
+        if file.type == "folder" then
+            UI:info(_("Retrieving folder ".. treePath.."/"..file.text .."…"))
+            local subtree = DropBox:getRemoteFilesMap(file.url,password,treePath.."/"..file.text)
+            for k,subf in pairs(subtree) do
+                tree[k]=subf
+            end
+        else
+            tree[treePath.."/"..file.text] = file
+        end
+	end
+    return tree
+end
+
+function DropBox:synchronize(item)
+    local function tableSlice(t, l,r)
+        if not r then r = #t end
+        local nt = {}
+        for i=l,r do
+            table.insert(nt,t[i])
+        end
+        return nt
+    end
+    local function stringSplit(s, sep)
+	--implem: https://stackoverflow.com/questions/1426954/split-string-in-lua
+        if sep == nil then
+                sep = "%s"
+        end
+        local t={}
+        for str in string.gmatch(s, "([^"..sep.."]+)") do
+                table.insert(t, str)
+        end
+        return t
+end
```

Comment:
Note that we have quite a few util functions.

http://koreader.rocks/doc/modules/util.html#gsplit